src/metadata/metadata.c

```diff
@@ -1118,20 +1118,42 @@ static void ocf_metadata_load_all_finish(ocf_pipeline_t pipeline,
 
 struct ocf_pipeline_arg ocf_metadata_load_all_args[] = {
 	OCF_PL_ARG_INT(metadata_segment_core_runtime),
-	OCF_PL_ARG_INT(metadata_segment_cleaning),
 	OCF_PL_ARG_INT(metadata_segment_lru),
 	OCF_PL_ARG_INT(metadata_segment_collision),
 	OCF_PL_ARG_INT(metadata_segment_list_info),
 	OCF_PL_ARG_INT(metadata_segment_hash),
 	OCF_PL_ARG_TERMINATOR(),
 };
 
+
+/*
+ * Predicate function checking whether segment is NULL used in
+ * pipeline conditional step
+ */
+static bool ocf_metadata_check_disabled_segment(ocf_pipeline_t pipeline,
+		void* priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_metadata_context *context = priv;
+	int segment_id = ocf_pipeline_arg_get_int(arg);
+	if (context->ctrl->segment[segment_id] != NULL)
+		return true;
+	return false;
+
+}
```

Comment:
I'm not sure if we really want to check for NULL segment (which is kind of side effect rather than actual API).
We could name this function `ocf_check_if_cleaner_disabled` (which is way more self-explaining and context-providing) and check the actual cache configuration.

I know where the temptation to make it more generic comes from, as we generate some amount of boilerplate, seemingly for nothing. But in any modern programming language the predicate would be a lambda function instead, and it would be as specific to the given context as possible (which in general is a good practice). So I'd rather follow the design pattern, even at cost of the boilerplate, rather than providing future-compatible, hackish filter function.

Reply:
I see, checking whether loading metadata cleaning section is needed through checking if cleaner disabled option is set
can be better way to go. Good point.